lib/Logger.js

```diff
@@ -20,72 +29,83 @@ class Logger {
     static get LogLevel() {
         return Object.keys(LogLevels).find(key => LogLevels[key] === LogLevel);
     }
+    static get LogLevels() {
+        return LogLevels;
+    }
     static set LogLevel(value) {
         if (LogLevels[value] === undefined) {
             throw "invalid log level '" + value + "', valid are '" + Object.keys(LogLevels).join("','") + "'";
         }
         LogLevel = LogLevels[value];
     }
+    static set LogFile(filename) {
+        if (LogFilename === filename) {
+            return;
+        }
+        if (LogFile) {
+            LogFile.close();
+        }
+        LogFilename = filename;
+        LogFile = fs.createWriteStream(LogFilename, LogFileOptions);
+    }
+    static get LogFile() {
+        return LogFilename;
+    }
+
+    static log(level, ...args) {
+        if (LogLevel <= LogLevels[level]) {
+            const logPrefix = BuildPrefix(level.toUpperCase());
+            // eslint-disable-next-line no-console
+            console.log(logPrefix, ...args);
```

Comment:
Before, the code was using `console.log`, `console.warn`, `console.error` etc which caused the log to be color-coded on supported setups without any additional logic as well as some stuff getting logged to stderr instead of stdout.

The latter wasn't utilized though.

The downside of using those methods was that the log was sometimes not chronologically sorted for some reason.